cpp/src/text/stemmer.cu

```diff
@@ -0,0 +1,266 @@
+    while (itr != d_str.end()) {
+      if (is_consonant(itr)) {
+        if (vowel_run) measure++;
+        vowel_run = false;
+      } else {
+        vowel_run = true;
+      }
+      ++itr;
+    }
+    return measure;
+  }
+};
+
+}  // namespace
+
+std::unique_ptr<cudf::column> porter_stemmer_measure(cudf::strings_column_view const& strings,
+                                                     cudaStream_t stream,
+                                                     rmm::mr::device_memory_resource* mr)
+{
+  if (strings.size() == 0) return cudf::make_empty_column(cudf::data_type{cudf::type_id::INT32});
+
+  // create empty output column
+  auto results = cudf::make_fixed_width_column(cudf::data_type{cudf::type_id::INT32},
+                                               strings.size(),
+                                               copy_bitmask(strings.parent(), stream, mr),
+                                               strings.null_count(),
+                                               stream,
+                                               mr);
+  // compute measures into output column
+  auto strings_column = cudf::column_device_view::create(strings.parent(), stream);
+  thrust::transform(rmm::exec_policy(stream)->on(stream),
+                    thrust::make_counting_iterator<cudf::size_type>(0),
+                    thrust::make_counting_iterator<cudf::size_type>(strings.size()),
+                    results->mutable_view().data<int32_t>(),
+                    porter_stemmer_measure_fn{*strings_column});
+  return results;
+}
+
+std::unique_ptr<cudf::column> is_letter(cudf::strings_column_view const& strings,
+                                        letter_type ltype,
+                                        cudf::column_view const& indices,
+                                        cudaStream_t stream,
+                                        rmm::mr::device_memory_resource* mr)
+{
+  return cudf::type_dispatcher(
+    indices.type(), dispatch_is_letter_fn{}, strings, ltype, indices, stream, mr);
+}
+
+}  // namespace detail
+
+// external APIs
+
+std::unique_ptr<cudf::column> is_letter(cudf::strings_column_view const& strings,
+                                        letter_type ltype,
+                                        cudf::size_type character_index,
+                                        rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  return detail::is_letter(
+    strings, ltype, thrust::make_constant_iterator<cudf::size_type>(character_index), 0, mr);
+}
+
+std::unique_ptr<cudf::column> is_letter(cudf::strings_column_view const& strings,
+                                        letter_type ltype,
+                                        cudf::column_view const& indices,
+                                        rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  return detail::is_letter(strings, ltype, indices, 0, mr);
+}
+
+/**
+ * @copydoc nvtext::porter_stemmer_measure
```

Comment:
Do we need this header over here ?

Reply:
Someone in a previous PR asked for this to be added. I figured it was harmless.